internal/lsp/server.go

```diff
@@ -62,36 +67,100 @@ type Server struct {
 	initializeParams *lsproto.InitializeParams
 	positionEncoding lsproto.PositionEncodingKind
 
+	watcheEnabled  bool
+	watcherID      int
+	watchers       map[project.WatcherHandle]struct{}
```

Comment:
Is this just a `core.Set[project.WatcherHandle]`?

Reply:
Oh yeah, I forgot we had something other than OrderedSet